donate/core/models.py

```diff
@@ -23,3 +28,69 @@ class LandingPage(Page):
         ImageChooserPanel('featured_image'),
         FieldPanel('intro'),
     ]
+
+
+class CampaignPage(Page):
+    template = 'pages/core/campaign_page.html'
+    parent_page_types = ['core.LandingPage']
+
+    hero_image = models.ForeignKey(
+        'wagtailimages.Image',
+        models.PROTECT,
+        related_name='+',
+    )
+    lead_text = models.CharField(max_length=800)
+
+    content_panels = Page.content_panels + [
+        ImageChooserPanel('hero_image'),
+        FieldPanel('lead_text'),
+        InlinePanel('donation_amounts', label='Donation amount overrides'),
+    ]
+
+
+class CampaignPageDonationAmount(models.Model):
+    campaign = ParentalKey(
+        'core.CampaignPage',
+        on_delete=models.CASCADE,
+        related_name='donation_amounts',
+    )
+    currency = models.CharField(
+        max_length=3,
+        choices=constants.CURRENCY_CHOICES,
+        blank=False,
+        db_index=True,
+    )
+
+    # Define 4 fields each for single and monthly preset amounts
+    preset_kwargs = {'max_digits': 7, 'decimal_places': 2}
+    single_1 = models.DecimalField(**preset_kwargs)
+    single_2 = models.DecimalField(**preset_kwargs)
+    single_3 = models.DecimalField(**preset_kwargs)
+    single_4 = models.DecimalField(**preset_kwargs)
+    monthly_1 = models.DecimalField(**preset_kwargs)
+    monthly_2 = models.DecimalField(**preset_kwargs)
+    monthly_3 = models.DecimalField(**preset_kwargs)
+    monthly_4 = models.DecimalField(**preset_kwargs)
```

Comment:
@solarissmoke Any thoughts about using a couple of simple `StreamField`s here instead, or are we happy always having exactly 4 single/monthly options each time?

Reply:
Interesting question. I could see us experimenting with more/fewer fields in the future; Is there any more/less significant overhead using a streamfield? (Sure, slight variations in database, UX of creating a campaign, but probably not major?)

Reply:
The only clear downside I can think of is that you'd have to manually click an 'Add' button a few times in order to enter the number of values you need (I'm not sure it's possibly to display 4 items on load). The additional computational cost of parsing the json string into a list of decimal values would be negligible.

It would make the values a little trickier to access directly in view/test code I supposed, but it would cut out a lot of duplication here, and in the template where they are output.

Reply:
Mmmm yes, testing. As long as we can still do that robustly, I'd be for the streamfield. (Obviously the CSS would have to be a bit forgiving, but that's nothing a little flexbox can't fix.)

Reply:
A streamfield would work from a data perspective - I had actually implemented this before reverting to a hard coded list. 

The reason I did that was because all the defaults are hard-coded, 4 each (and there is a large number of currencies). This means that if you wanted to change to say 5 options, you would either have to do it for *all* currencies, or allow different numbers of options for different currencies, and a proper way of handling showing/hiding extra options in JS on the front end. This is all doable but adds complexity to that logic.

What do you think @alanmoo ? Open to going either route.

Reply:
Ah, great point. That makes me lean towards leaving it as-is, but I’ll confirm with Will tomorrow (he ended up needing today off too).

Reply:
Will was very interested in the Streamfield route, so let's go with that. I'm happy to hop on a call to hash out the details on this if needed (or at least understand what decisions need to be made) as early as tomorrow at 9AM EDT, just send me an invite.

Reply:
As per discussion yesterday, we're going with a streamfield.